modules/_unstable/README.md

```diff
@@ -0,0 +1,59 @@
+# Canvas Kit Unstable
+
+This is a group of unstable/work in progress components.
+
+Since we have a rather rigorous process for getting components in at a production level, it can be
+valuable to make them available earlier for teams who need them during development. Many of our
+components are contributions from internal product teams. While these components can suit the
+contributing product's use case, it takes time to iterate on the API/functionality as we evaluate
+them in a more global context. The Unstable modules allow us to do that as needed.
+
+## Breaking Changes
+
+Due to the nature of this bundle of components, we will be continuously iterating, likely making
+breaking changes. The goal of this module is to work unconstrained by our design system's semantic
+versioning. This means that **breaking changes can be deployed to unstable modules at any time and
+it will not trigger a major version bump across our other modules**. Since we use a fixed version
+number for all components, all changes/additions in this bundle will never result in a major version
+bump across our component library.
+
+**By consuming any of these modules, you are indicating that you are prepared for the consequences
+of updating and ultimately performing a migration when components are promoted to a stable
+version.**
+
+**These components are purely opt in**, so they are not included in the universal modules
+`@workday/canvas-kit-css` and `@workday/canvas-kit-react`.
+
+## Module Structure
+
+```
+/modules
+  ...
+  /_unstable
+    /new-component
+      /css
+        /* Standard CSS component structure */
+      /react
+        /* Standard React component structure */
+```
+
+## Creating an Unstable Module
+
+1. Run `yarn create-module`
+2. When asked `Is this an unstable component? [Y/n]`, enter `y` or `Y`.
+3. Your new module will be generated in accordance with the file structure above.
+
+## Migrating a Component into Unstable
+
+1. Copy it into the above module structure
+2. Add a warning to the README:
+   > <a href="https://github.com/Workday/canvas-kit/tree/master/modules/_unstable/README.md">
+   >   <img src="https://img.shields.io/badge/UNSTABLE-alpha-orange" alt="UNSTABLE: Alpha" />
+   > </a>  This component is work in progress and currently in pre-release.
+3. Ensure a prerelease version is applied that matches the badge above
+
+## Deploying
+
+- Components in Unstable are given prerelease versions. Their general version will align with the
+  rest of the library (e.g. `3.0.0-alpha.0`)
+- Before deploying, ensure all unstable modules have a prerelease tag (`alpha` or `beta`).
```

Comment:
How do we enforce this?

We’ve just been running `yarn lerna version --no-git-tag-version --no-push --no-changelog prerelease` for releases I believe (@lychyi , correct me if this has changed - you’ve done the last couple). This works when everything is in prerelease, but when we do a final version (without `prerelease`), we’ll have to manually version each module in Unstable to add the prerelease tags.

Should I add a new yarn version command that bumps all modules and then versions everything in `modules/_unstable` as `alpha`? What if we want to support `beta`? This feels like it should be a CLI tool. However, for the sake of getting this PR in ASAP, I'm thinking we do this manually for now and add an issue to work on a script for this.

Reply:
Prereleases can be tricky. If we release a `3.0.0` tomorrow, then `canvas-kit@^3.0.0` will no longer match pre-releases even if they occur later. We can use tags, however. By default, `npm publish` will do `npm publish --tag latest` including pre-releases. If we want to publish intermediate changes, we'd probably want more like a `-build-{hash}` and publish the tag as `next` or `master` or `unstable` or something.

I don't think we want to use `alpha` or `beta`, but more like `build`. No promises about `unstable` other than builds, right?

Reply:
@anicholls We don't run that command explicitly anymore. Here were the tweaks from last time:
* Removed `--no-changelog`. It not necessary without `--conventional-commits` ([link](https://github.com/lerna/lerna/tree/master/commands/version#--no-changelog))
* The semver bump positional arg: `prerelease` is now something that someone must select manually when using `yarn bump` (this way this command can be used for both prerelease bumps and regular semantic bumps)
* All options for `lerna version` are all moved into `lerna.json`

The `prerelease` tag is now specified in `travis.yml` as a [build step](https://github.com/Workday/canvas-kit/blob/master/.travis.yml#L30) as a `--pre-dist-tag` option for `lerna publish`. This allows publish to tag things as prerelease only if they have a prerelease version (`*-alpha.X` for example). Everything else should default to `latest` on publish.

Reply:
This isn't about prereleases of any stable modules (or the bundle modules). Essentially we are talking about fixed versioning for everything _not_ in `_unstable`. Then manually versioning the items in `_unstable`. However, I'm not sure this is really possible without going back to independent versioning... (which we really don't want to do)

IMO, a warning in the readme, the folder the module is in within the source repo and an npm tag is super easy to miss from the consumer's perspective...

Reply:
So after a very long discussion about the best direction to go, we arrived at this:

## Short term
- We decided to call this area "Canvas Kit Labs". It's more concise and feels more positive than "unstable"
- All unstable modules will move into a `_labs` folder
- All unstable modules in this folder will have `-labs` put in their component name (i.e. `@workday/canvas-kit-labs-react-menu`. We are unconcerned about the pollution in the namespace because they're very long and names are cheap. It is extremely unlikely we will need this namespace in the future. This will be seen anywhere you see the version and will offer similar clarity to a prerelease version. It avoids the need for going back to independent versioning (for now).

## Long term
Consider one of two options:
- Switch back to "independent" versioning in Lerna, but write a Lerna integration script that keeps the majority of our components on a fixed version, while allowing us to manage select modules independently (e.g. unstable components)
- If we move to a slash-style import system (a single bundle, i.e. `@workday/canvas-kit/react/button`), it would make this very easy (e.g. `@workday/canvas-kit/labs/react/menu`)